src/composables/node/useNodePricing.ts

```diff
@@ -168,7 +168,40 @@ const byteDanceVideoPricingCalculator = (node: LGraphNode): string => {
     ? minStr
     : `$${minCost.toFixed(2)}-$${maxCost.toFixed(2)}/Run`
 }
+const sora2PricingCalculator: PricingFunction = (node: LGraphNode): string => {
+  const modelW = node.widgets?.find((w) => w.name === 'model') as IComboWidget
+  const durationW = node.widgets?.find(
+    (w) => w.name === 'duration' || w.name === 'duration_s'
+  ) as IComboWidget
+  const sizeW = node.widgets?.find((w) => w.name === 'size') as IComboWidget
+
+  // precise missing-widget messages (fixes the failing test)
+  if (!modelW || !durationW) return 'Set model, duration & size'
+  if (!sizeW) return 'Set size (720x1280, 1280x720, 1024x1792, 1792x1024)'
+
+  const model = String(modelW.value ?? '').toLowerCase()
+  const duration = Number(durationW.value)
+  const size = String(sizeW.value ?? '').toLowerCase()
+
+  if (!duration || Number.isNaN(duration)) return 'Set duration (4/8/12)'
+  if (!size) return 'Set size (720x1280, 1280x720, 1024x1792, 1792x1024)'
+
+  const SORA720 = ['720x1280', '1280x720']
+  const SORA_PRO_HIRES = ['1024x1792', '1792x1024']
+
+  if (model.includes('sora-2-pro')) {
+    let perSec: number | null = null
+    if (SORA_PRO_HIRES.includes(size)) perSec = 0.5
+    else if (SORA720.includes(size)) perSec = 0.3
+    else return 'Size must be 720x1280, 1280x720, 1024x1792, or 1792x1024'
+    return `$${(perSec * duration).toFixed(2)}/Run`
+  }
 
+  // plain sora-2: 720p only
+  if (!SORA720.includes(size))
+    return 'sora-2 supports only 720x1280 or 1280x720'
```

Comment:
I'd love another pure function here that just does the validation and returns `string | undefined` where the string is the error message